src/test/java/com/google/sps/TripServletTest.java

```diff
@@ -0,0 +1,131 @@
+    when(mockRequest.optimizeWaypoints(anyBoolean())).thenReturn(mockRequest);
+    when(mockRequest.mode(any())).thenReturn(mockRequest);
+
+    GeoApiContext mockGeoApiContext = mock(GeoApiContext.class);
+
+    String[] waypoints = new String[]{ "MoPOP, 5th Avenue North, Seattle, WA, USA",
+                                    "Space Needle, Broad Street, Seattle, WA, USA",
+                                    "Alki Beach, Seattle, WA, USA"};
+    
+    String origin = "The Westin Bellevue, Bellevue Way Northeast, Bellevue, WA, USA";
+
+    DirectionsApiRequest request = TripServlet.generateDirectionsRequest(origin, waypoints, mockGeoApiContext);
+
+    // Verify that proper methods and parameters are called to generate directionsRequest.
+    verify(mockRequest).origin(origin);
+    verify(mockRequest).destination(origin);
+    verify(mockRequest).waypoints(waypoints);
+    verify(mockRequest).optimizeWaypoints(true);
+    verify(mockRequest).mode(TravelMode.DRIVING);
+  }
+
+  // Test waypoint order parsing from a DirectionsResult object.
+  @Test
+  public void getOrderedWaypointsTest() {
+    // Manually create directions result object.
+    DirectionsResult dirResult = new DirectionsResult();
+    dirResult.routes = new DirectionsRoute[1];
+    dirResult.routes[0] = new DirectionsRoute();
+    dirResult.routes[0].waypointOrder = new int[]{ 1, 2, 0 };
+    
+    // Manually create user input.
+    String[] pois = new String[]{"Alki Beach, Seattle, WA, USA",
+                                  "MoPOP, 5th Avenue North, Seattle, WA, USA",
+                                  "Space Needle, Broad Street, Seattle, WA, USA"};
+
+    List<String> orderedWaypoints = TripServlet.getOrderedWaypoints(dirResult, pois);
+
+    List<String> expectedWaypointOrder = new ArrayList<>();
+    expectedWaypointOrder.add("MoPOP, 5th Avenue North, Seattle, WA, USA");
+    expectedWaypointOrder.add("Space Needle, Broad Street, Seattle, WA, USA");
+    expectedWaypointOrder.add("Alki Beach, Seattle, WA, USA");
+
+    Assert.assertEquals(orderedWaypoints, expectedWaypointOrder);
+  }
+
+  // Test travel time parsing from a DirectionsResult object.
+  @Test
+  public void getTravelTimesTest() {
+    // Manually create directions result object
+    DirectionsResult dirResult = new DirectionsResult();
+    dirResult.routes = new DirectionsRoute[1];
+    dirResult.routes[0] = new DirectionsRoute();
+    dirResult.routes[0].legs = new DirectionsLeg[4];
+    for (int i = 0; i < dirResult.routes[0].legs.length; i++) {
+      dirResult.routes[0].legs[i] = new DirectionsLeg();
+      dirResult.routes[0].legs[i].duration = new Duration();
+    }
+    dirResult.routes[0].legs[0].duration.inSeconds = 1080;
+    dirResult.routes[0].legs[1].duration.inSeconds = 60;
+    dirResult.routes[0].legs[2].duration.inSeconds = 1500;
+    dirResult.routes[0].legs[3].duration.inSeconds = 2100;
+    
+    List<Integer> travelTimes  = TripServlet.getTravelTimes(dirResult);
```

Comment:
personal preference: Since your calling the other `expected` maybe call this one `actualTravelTimes`